### ISSUE TITLE
Test data post-migration

### ISSUE THREAD

**Issue by strogonoff** [isogr/register-system-transition#30]
The goal is to ensure that data output by the new system (currently https://geodetic-v2.isotc211.org/) exactly matches the data output by the old system (https://geodetic-static.isotc211.org/).

cc @stefanomunarini, @ribose-jeffreylau if anyone has ideas.

I’ve been researching various snapshot testing (+ optional visual comparison) approaches (e.g., [with playwright](https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-screenshot-1)), but I wonder if that’s overkill.

**Comment by ribose-jeffreylau** [comment]
Currently, PDF files downloaded from the Java-based system contains retrieval timestamps.  Just a small bit of detail to be aware of.

**Comment by ribose-jeffreylau** [comment]
Questions regarding snapshot (which I'll call screenshot from now on, to avoid potential confusion) testing:

- Is achieving pixel perfect necessary for screenshot testing to work?

Currently, the item class index pages you see in the Java-based (and hence also in the static *snapshot*) website are all rendered with a data JSON file behind the scene.  In the snapshot directory, you will be able to find various JSON files called `*containedItem`.  These files may help in determining differences in number of items in each item class.

For individual item pages, may need to extract just the text from the relevant elements and go from there.
If we want more semantics from the pages though, I believe we can use the fact that most fields are enclosed in `<p id="...fieldName...">` tags which would help with initial data cleaning.

*E.g.,* `<input disabled="disabled" id="domainOfValidity.geographicBoundingBoxes0.northBoundLatitude" name="domainOfValidity.geographicBoundingBoxes[0].northBoundLatitude" value="62...`

**Comment by stefanomunarini** [comment]
> I’ve been researching various snapshot testing (+ optional visual comparison) approaches (e.g., [with playwright](https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-screenshot-1)), but I wonder if that’s overkill.

How were you thinking to achieve this @strogonoff ? Would this include exporting a PDF from Paneron and comparing the two? (I cannot see if this functionality is already in place, cause at the moment I cannot access https://geodetic-v2.isotc211.org)

**Comment by stefanomunarini** [comment]
> Questions regarding snapshot (which I'll call screenshot from now on, to avoid potential confusion) testing:
> 
> * Is achieving pixel perfect necessary for screenshot testing to work?
> 
> Currently, the item class index pages you see in the Java-based (and hence also in the static _snapshot_) website are all rendered with a data JSON file behind the scene. In the snapshot directory, you will be able to find various JSON files called `*containedItem`. These files may help in determining differences in number of items in each item class.

This is great @ribose-jeffreylau , we could use part of this data (e.g. management proposal info) to remove some overhead while retrieving data for individual item pages using HTML selectors

**Comment by ribose-jeffreylau** [comment]
I have neutralized the PDF retrieval timestamps to `1970-01-0T00:00:00` which may ease PDF comparison...

- changes made to `scripts/scrape`: f34b7d649dd969c8b885dd2439405001417bb7b8
- changes made to the PDF themselves: 4fd849b89eb44d2f56ca057eb9f2e6a6cd42826c

**Comment by ribose-jeffreylau** [comment]
@stefanomunarini There is not yet a PDF export feature in the new system.

BTW what errors are you getting when accessing v2 site?

**Comment by stefanomunarini** [comment]
> BTW what errors are you getting when accessing v2 site?
<img width="1512" alt="Screenshot 2024-02-06 at 10 54 23" src="https://github.com/isogr/register-system-transition/assets/4653319/922f5381-68c2-4e7d-8da0-e38ba7d84acc">

**Comment by stefanomunarini** [comment]
> > BTW what errors are you getting when accessing v2 site?
> 
> <img alt="Screenshot 2024-02-06 at 10 54 23" width="1512" src="https://private-user-images.githubusercontent.com/4653319/302597978-922f5381-68c2-4e7d-8da0-e38ba7d84acc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyMTcxOTQsIm5iZiI6MTcwNzIxNjg5NCwicGF0aCI6Ii80NjUzMzE5LzMwMjU5Nzk3OC05MjJmNTM4MS02OGMyLTRlN2QtOGRhMC1lMzhiYTdkODRhY2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDZUMTA1NDU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGY3OGE4Yjc1MDhjNTQ3YjdmNzkzMTIwY2QxZDE5N2MzN2U2MTNlZGM4ZjI4YTljODg4NDI1MGI2M2JlZmY5NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ZAFytH3mMTAwiMORH-qCiX1FQJlzrbs5Et-qxzKerqQ">

@ribose-jeffreylau

**Comment by ribose-jeffreylau** [comment]
@stefanomunarini Do you have Safari v16?  It should work in v17.  Or try other browsers.

**Comment by stefanomunarini** [comment]
> @stefanomunarini Do you have Safari v16? It should work in v17. Or try other browsers.

Thanks @ribose-jeffreylau . It works in Safari v16.6

**Comment by strogonoff** [comment]
> [@stefanomunarini](https://github.com/stefanomunarini) Do you have Safari v16? It should work in v17. Or try other browsers.

Yes, it also works with year old Firefox and Chrome. Only Safari has to be new?
